src/main/kotlin/study/Calculator.kt

```diff
@@ -0,0 +1,113 @@
+package study
+
+class Calculator(
+    private var value: Int = 0,
+    private var result: Double = 0.0,
+
+) {
+    companion object {
+        private const val CHECK_ZERO = 0
+        private const val CHECK_POINT_ONE = 1
+        private const val PROCESS_KEYWORD_NUMBER = "number"
+        private const val PROCESS_KEYWORD_OPERATOR = "operator"
+
+        private val regex1 = "\\s+".toRegex()
+        private val regex2 = "\\D+".toRegex()
+        private val regex3 = "[\\+\\-\\*\\/]".toRegex()
+    }
+
+    fun getResult(): Any {
+        return if (! (this.result % CHECK_POINT_ONE > CHECK_ZERO)) {
+            this.result.toInt()
+        } else {
+            this.result
+        }
+    }
```

Comment:
```suggestion
    fun getResult() = if (!(this.result % CHECK_POINT_ONE > CHECK_ZERO)) {
            this.result.toInt()
        } else {
            this.result
        }
```

코틀린에서는 조건문이 함수식으로도 활용이 가능합니다. 😊